libsolidity/ast/Types.cpp

```diff
@@ -447,7 +447,7 @@ MemberList::MemberMap Type::boundFunctions(Type const& _type, ASTNode const& _sc
 		)
 			continue;
 		auto const& library = dynamic_cast<ContractDefinition const&>(
-			*ufd->libraryName().annotation().referencedDeclaration
+			*ufd->libraryName().pathNode()->annotation().referencedDeclaration
```

Comment:
Maybe it would make sense to create a shortcut function inside IdentifierPath to return `pathNode()->annotation().referencedDeclaration`?

Reply:
Do you mean inside `IdentifierPath` or inside `UserDefinedTypeName`? If you mean the first the above example would become
```
*ufd->libraryName().pathNode()->declaration()
```